client/src/entrypoints/admin/preview-panel.js

```diff
@@ -1,11 +1,66 @@
+import axe from 'axe-core';
+import {
+  getAxeConfiguration,
+  renderA11YResults,
+} from '../../includes/a11y-result';
 import { WAGTAIL_CONFIG } from '../../config/wagtailConfig';
 import { debounce } from '../../utils/debounce';
 import { gettext } from '../../utils/gettext';
 
+const runAccessibilityChecks = async (newTabButton) => {
```

Comment:
I think I'd prefer we accept `onClickSelector` and just pass it to `renderA11yResults`, and do something like `runAccessibilityChecks(() => newTabButton.click())` in the `initPreview()` code, just so that we don't rely on a variable inside the `initPreview` code.

It's pretty minor and I don't have strong feelings leaving this as-is though